### DISCUSSION TITLE
Can DLLs be Added and Configured in Ghidra on Linux, or Does It Require a Windows Environment?

### DISCUSSION THREAD

**Discussion by 0xh3xa** [NationalSecurityAgency/ghidra#6911] [Q&A]
I'm reversing PE files on Linux using Ghidra 11.1.2. Is it possible to add DLLs and set paths within Ghidra on this environment, or does it require a Windows host to function properly?

```
------------------------------------------------

Searching 1 local path for library GDIPLUS.DLL...
Library not found.
Searching 6 system paths for library GDIPLUS.DLL...
Library not found.
------------------------------------------------

Searching 1 local path for library KERNEL32.DLL...
Library not found.
Searching 6 system paths for library KERNEL32.DLL...
Library not found.
------------------------------------------------

Searching 1 local path for library MSACM32.DLL...
Library not found.
Searching 6 system paths for library MSACM32.DLL...
Library not found.
------------------------------------------------

Searching 1 local path for library MSVFW32.DLL...
Library not found.
Searching 6 system paths for library MSVFW32.DLL...
Library not found.
------------------------------------------------

Searching 1 local path for library SHELL32.DLL...
Library not found.
Searching 6 system paths for library SHELL32.DLL...
Library not found.
------------------------------------------------

Searching 1 local path for library USER32.DLL...
Library not found.
Searching 6 system paths for library USER32.DLL...
Library not found.
------------------------------------------------

Linking the External Programs of 'file.exe' to imported libraries...
  [KERNEL32.DLL] -> not found in project
  [USER32.DLL] -> not found in project
  [SHELL32.DLL] -> not found in project
  [MSVFW32.DLL] -> not found in project
  [GDIPLUS.DLL] -> not found in project
  [MSACM32.DLL] -> not found in project
------------------------------------------------
```

**Comment by ryanmkurtz** [comment]
It is possible.  When you import your program, hit the `Options...` button in the Import dialog.  Check `Load Libraries From Disk`, and then click the `Edit Paths` button next to `Library Search Paths` to configure where it looks.  Some of this exact wording might not be identical in whatever version of Ghidra you are using...it has recently changed.  But the basic concept still appies.

**Comment by 0xh3xa** [comment]
Thank you for your response. Can I achieve the same result with analyzeHeadless, and if so, how?
I have already checked the arguments from https://static.grumpycoder.net/pixel/support/analyzeHeadlessREADME.html

**Reply by ryanmkurtz** [reply]
You can.  What version of Ghidra are you using?

**Comment by 0xh3xa** [comment]
@ryanmkurtz I am using `Ghidra 11.1.2`

**Reply by ryanmkurtz** [reply]
For that version, you'd do something like 
```
./analyzeHeadless <path/to/project> <project_name> -import <base_program> -loader PeLoader -loader-loadSystemLibraries true
```

For Ghidra 11.1.2, you are going to have to use the GUI to set and save the system library search paths.  Ghidra 11.2 (in development but nearing a release) will have a new headless command line argument to specific the search paths on the command line.

**Comment by 0xh3xa** [comment]
Ok, thanks for your support!
